<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"          default="OAK-D"   /> <!-- 'zed' or 'zedm' -->
    <arg name="tf_prefix"             default="oak"         />
    <arg name="base_frame"            default="oak-d_frame" />
    <arg name="parent_frame"          default="oak-d-base-frame" />

    <arg name="cam_pos_x"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="1" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="1.57" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://depthai_examples/params/camera" />
    <arg name="sync_nn"               default="true"/>
    <arg name="subpixel"              default="true"/>
    <arg name="confidence"            default="200" />
    <arg name="LRchecktresh"          default="5"   />
    <arg name="monoResolution"        default="400p"/> <!-- '720p', '800p', 400p' for OAK-D & '480p' for OAK-D-Lite -->
    <arg name="nnName"                default="yolov4_tiny_coco_416x416_openvino_2021.4_6shave_bgr.blob"         />
    <arg name="resourceBaseFolder"    default="$(find my_depthai)/resources" />

    <arg name="enableRviz"            default="true"/>
    <arg name="enableMarkerPublish"   default="false" />

    <include file="$(find my_depthai)/launch/urdf_ex.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="parent_frame"    value="$(arg  parent_frame)"/>
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="tf_prefix"       value="$(arg  tf_prefix)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"   />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"   />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"   />
        <arg name="cam_roll"        value="$(arg  cam_roll)"    />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"   />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"     />
    </include>

    <node pkg="tf2_ros" type="static_transform_publisher" name="oak_broadcaster" args="0 0 0 0 0 0 1 world oak-d-base-frame" />


    <node name="yolov4_publisher" pkg="depthai_examples" type="yolov4_spatial_node" output="screen" required="true">
        <param name="tf_prefix"           value="$(arg tf_prefix)"     />
        <param name="camera_param_uri"    value="$(arg camera_param_uri)"/>
        <param name="sync_nn"             value="$(arg sync_nn)"/>
        <param name="subpixel"            value="$(arg subpixel)"/>
        <param name="confidence"          value="$(arg confidence)"/>
        <param name="LRchecktresh"        value="$(arg LRchecktresh)"/>
        <param name="monoResolution"      value="$(arg monoResolution)"/>
        <param name="nnName"              value="$(arg nnName)"/>
        <param name="resourceBaseFolder"  value="$(arg resourceBaseFolder)"/>
    </node>

    <group if="$(eval arg('enableRviz') == true)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find my_depthai)/rviz/yolo4.rviz" />
      <!--
      <node if="$(eval arg('depth_aligned') == true)" type="rviz" name="rviz" pkg="rviz" args="-d $(find my_depthai)/rviz/stereoInertialDepthAlign_w_manipulator.rviz" />
      <node if="$(eval arg('depth_aligned') == false)" type="rviz" name="rviz" pkg="rviz" args="-d $(find my_depthai)/rviz/stereoInertialROS1_w_manipulator.rviz" />
    -->
    </group>

</launch>
